index/scorch/segment/segment.go

```diff
@@ -55,7 +55,7 @@ type DictionaryIterator interface {
 }
 
 type PostingsList interface {
-	Iterator() PostingsIterator
+	Iterator(includeFreq, includeNorm, includeLocations bool) PostingsIterator
```

Comment:
i'm ok either way on this one.  in general the struct might be nicer, but we've already got these individual flags higher up in the API, so doesn't feel like we're making anything worse...

Reply:
>> one doubt, can these parameters be comprised in a struct arg so that any future changes can be easily accommodable in future, and feels like that looks cleaner?

> i'm ok either way on this one. ...

yah, i figure let's push this one in as-is, and if cleanup or struct'ing is needed, a future PR would be the way to go